src/plot_api/plot_api.js

```diff
@@ -941,6 +941,7 @@ Plotly.redraw = function(gd) {
 Plotly.newPlot = function (gd, data, layout, config) {
     gd = getGraphDiv(gd);
     plots.purge(gd);
+    gd.removeAllListeners();
```

Comment:
I'd would put this in `plots.purge` 

and maybe (cc @alexcjohnson ) we could expose `purge` on `Plotly`? It sounds to me like something users might want to use.